src/ir/possible-contents.h

```diff
@@ -265,7 +267,9 @@ class PossibleContents {
 
   // Returns whether the relevant cone for this, as computed by getCone(), is of
   // full size, that is, includes all subtypes.
-  bool hasFullCone() const { return getCone().depth == FullDepth; }
+  bool hasFullCone() const {
+    return getCone().depth == (getType().isExact() ? 0 : FullDepth);
+  }
```

Comment:
Does this need to change? It seems odd to consider a code of depth 0 as "full".

Reply:
Yes, otherwise a bunch of assertions expecting full cone types started failing. I think it makes sense to make cone creation and inspection consistent in treating depth-0 cones for exact types as "full," so I think this is a better solution than going and changing those assertions.

Reply:
I'd like to understand this better - which assertions were those?

Reply:
Hmm, that assert looks suspicious to me... I'd rather investigate that first, but I don't feel strongly.

If you want to land this quickly to unblock other stuff, how about doing the reverse: modify the assert to allow either a full cone or an exact type, so we don't need to modify hasFullCone? I can also take a look at the PossibleContents code if you want.

Reply:
No rush to land this. I can reorder things to avoid getting blocked. I do think it would be weird if a value created by `fullConeType` did not satisfy `hasFullCone` or `isFullCone`, though.

Reply:
Yeah, we should rename `fullConeType` probably. Really what it does now is create a cone based on the type. Just `coneType` might be enough.